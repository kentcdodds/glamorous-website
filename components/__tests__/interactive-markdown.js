import React from 'react'
import {mount} from 'enzyme'
import toJson from 'enzyme-to-json'
import {matcher, serializer} from 'jest-glamor-react'
import interactiveMarkdown from '../interactive-markdown'

expect.addSnapshotSerializer(serializer)
expect.extend(matcher)

function flushAllPromises() {
  return new Promise(resolve => setTimeout(resolve, 200))
}

test('parses interactive pragmas into CodePreviews', async () => {
  const markdown = `
    # hello world

    I am some content
    ~~~interactive {clickToRender: true, summary: 'I am the summary'}
    render(<button onClick={() => alert('Hello World')}>Hello World</button>)
    ~~~

    I am some more content
  `
  const elements = interactiveMarkdown(markdown)
  const wrapper = mount(<div>{elements}</div>)
  expect(toJson(wrapper)).toMatchSnapshotWithGlamor()
  wrapper.find('details').simulate('click')
  wrapper.find('details').first().node.open = true
  await flushAllPromises()
  expect(toJson(wrapper)).toMatchSnapshotWithGlamor()
})
