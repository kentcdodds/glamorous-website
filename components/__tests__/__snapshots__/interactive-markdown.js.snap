// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parses interactive pragmas into CodePreviews 1`] = `
<div>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<h1>hello world</h1>
      <p>I am some content</p>
      ",
      }
    }
  />
  <ClickToRenderCodePreview
    clickToRender={true}
    code="render(<button onClick={() => alert('Hello World')}>Hello World</button>)"
    component={[Function]}
    summary="I am the summary"
  >
    <ClickToRender
      component={[Function]}
      props={
        Object {
          "clickToRender": true,
          "code": "render(<button onClick={() => alert('Hello World')}>Hello World</button>)",
          "component": [Function],
          "summary": "I am the summary",
        }
      }
      summary="I am the summary"
    >
      <details
        onClick={[Function]}
      >
        <summary>
          I am the summary
        </summary>
      </details>
    </ClickToRender>
  </ClickToRenderCodePreview>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "
      <p>I am some more content
      </p>
      ",
      }
    }
  />
</div>
`;

exports[`parses interactive pragmas into CodePreviews 2`] = `
<div>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<h1>hello world</h1>
      <p>I am some content</p>
      ",
      }
    }
  />
  <ClickToRenderCodePreview
    clickToRender={true}
    code="render(<button onClick={() => alert('Hello World')}>Hello World</button>)"
    component={[Function]}
    summary="I am the summary"
  >
    <ClickToRender
      component={[Function]}
      props={
        Object {
          "clickToRender": true,
          "code": "render(<button onClick={() => alert('Hello World')}>Hello World</button>)",
          "component": [Function],
          "summary": "I am the summary",
        }
      }
      summary="I am the summary"
    >
      <details
        onClick={[Function]}
      >
        <summary>
          I am the summary
        </summary>
      </details>
    </ClickToRender>
  </ClickToRenderCodePreview>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "
      <p>I am some more content
      </p>
      ",
      }
    }
  />
</div>
`;
